packages/misc/src/lib/helper/session-sigs-validator.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  AuthSig,
+  Capability,
+  ParsedSessionMessage,
+  ParsedSignedMessage,
+  SessionSigsMap,
+} from '@lit-protocol/types';
+
+interface ValidationResult {
+  isValid: boolean;
+  errors: string[];
+}
+
+// Function to parse a signedMessage string into an object
+function parseSignedMessage(signedMessage: string): ParsedSignedMessage {
+  const lines = signedMessage.split('\n');
+  const parsedData: ParsedSignedMessage = {};
+  let currentKey: string | null = null as string | null;
+  let currentValue = '';
+
+  lines.forEach((line) => {
+    // Match lines with 'Key: Value' pattern
+    const keyValueMatch = line.match(/^([^:]+):\s*(.*)$/);
+
+    if (keyValueMatch) {
+      // Save the previous key-value pair
```

Comment:
Iteration saving is on the previous value because of the case of lists where it appends until a next value occurs, did I get that right?

Reply:
yea, so let's say the given string, the first line is:

iteration: 0
line: `localhost wants you to sign in with your Ethereum account:`

it matches the KV pattern, so set `currentKey` to `localhost wants you to sign in with your Ethereum account` and `currentValue` to an empty string.

iteration: 1
line: `0x70997970C51812dc3A010C7d01b50e0d17dc79C8`

it doesn't match anything, so concatenate the line to `currentValue` 

iteration: 2
it's an empty line, so skipping it

iteration 3:
line: `This is a test statement. You can put anything you want here...`

it matches the KV pattern, but **before setting the new** currentKey and currentValue, the code saves the previous  KV pair:

```
if (currentKey !== null) {
  parsedData[currentKey.trim()] = currentValue.trim();
}
```

so:
```
Saves parsedData['localhost wants you to sign in with your Ethereum account'] as '0x70997970C51812dc3A010C7d01b50e0d17dc79C8'.
```

Though, I think the code is overly complicated, we might want to re-visit this when we could use the SIWE-Recap library

Reply:
Thanks for the explanation!